public/app/plugins/panel/geomap/style/markers.ts

```diff
@@ -319,34 +313,75 @@ export function getMarkerAsPath(shape?: string): string | undefined {
-    symbol: {
-      symbolType: 'circle',
-      size: ['get', 'size', 'number'],
-      color: ['color', ['get', 'red'], ['get', 'green'], ['get', 'blue']],
-      offset: ['array', ['get', 'offsetX'], ['get', 'offsetY']],
-      rotation: ['get', 'rotation', 'number'],
-      opacity: ['get', 'opacity', 'number'],
-    },
-  };
-  // set symbolType and src if a symbol is provided
-  if (symbol && symbolStyle.symbol) {
-    const imageString = 'image';
-    const symbolType = Object.keys(WebGLRegularShapes).find((key) => WebGLRegularShapes[key] === symbol) ?? imageString;
-    symbolStyle.symbol = { ...symbolStyle.symbol, symbolType };
-    if (symbolType === imageString) {
-      const backgroundOpacity = opacity === 0 ? 0 : 0.1 / (opacity ?? 1);
-      symbolStyle.symbol = {
-        ...symbolStyle.symbol,
-        src: await prepareSVG(getPublicOrAbsoluteUrl(symbol), undefined, backgroundOpacity),
-      };
-    }
+// Common expressions used across different style types
+export const colorExpression = ['color', ['get', 'red'], ['get', 'green'], ['get', 'blue'], ['get', 'opacity']];
+export const sizeExpression = ['get', 'size'];
+export const opacityExpression = ['get', 'opacity'];
+export const rotationExpression = ['get', 'rotation'];
+
+// Base style for regular shapes
+export const baseShapeStyle = {
+  'shape-radius': sizeExpression,
+  'shape-fill-color': colorExpression,
+  'shape-stroke-color': colorExpression,
+  'shape-stroke-width': 1,
+  'shape-opacity': opacityExpression,
+  'shape-rotation': rotationExpression,
+};
+
+// Base style for circles
+export const baseCircleStyle = {
+  'circle-radius': sizeExpression,
+  'circle-fill-color': colorExpression,
+  'circle-stroke-color': colorExpression,
+  'circle-stroke-width': 1,
+  'circle-opacity': opacityExpression,
+};
+
+// Returns style configuration for WebGL markers
+export async function getWebGLStyle(symbol?: string, opacity?: number): Promise<FlatStyle> {
+  // Handle circle explicitly (before generic SVG check)
+  if (symbol === MarkerShapePath.circle) {
+    return baseCircleStyle;
+  }
+
+  // Handle square as WebGL regular shape
+  if (symbol === MarkerShapePath.square) {
+    return {
+      ...baseShapeStyle,
+      'shape-points': 4,
+      'shape-angle': Math.PI / 4,
+    };
+  }
+
+  // Handle triangle as WebGL regular shape
+  if (symbol === MarkerShapePath.triangle) {
+    return {
+      ...baseShapeStyle,
+      'shape-points': 3,
+      'shape-angle': 0,
+    };
+  }
+
+  // Handle custom SVG symbols and other shapes as icons
+  if (symbol && symbol.endsWith('.svg')) {
+    const backgroundOpacity = opacity === 0 ? 0 : 0.1 / (opacity ?? 1);
+    return {
+      'icon-src': await prepareSVG(getPublicOrAbsoluteUrl(symbol), undefined, backgroundOpacity),
+      'icon-width': sizeExpression,
+      'icon-height': sizeExpression,
+      'icon-opacity': opacityExpression,
+      'icon-rotation': rotationExpression,
+      'icon-offset': [
```

Comment:
Switching over to `FlatStyle` removes support for offset. Need to find a workaround.